src/Apps/Order/Routes/Shipping2/Hooks/useBackToFulfillmentDetails.tsx

```diff
@@ -1,64 +0,0 @@
-import { usePrevious } from "@artsy/palette"
-import { useShippingContext } from "Apps/Order/Routes/Shipping2/Hooks/useShippingContext"
-import { matchAddressFields } from "Apps/Order/Routes/Shipping2/Utils/shippingUtils"
-import { extractNodes } from "Utils/extractNodes"
-import { Shipping2_me$data } from "__generated__/Shipping2_me.graphql"
-import { useEffect } from "react"
-
-/**
- * Go back to fulfillment details stage if the user edits the address or
- * deletes a saved address.
- */
-export const useBackToFullfillmentDetails = (me: Shipping2_me$data) => {
```

Comment:
Moved this logic into a wrapper around specific `handleChange` callbacks that should trigger the change. It's cleaner, less code and easier to trace.